storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java

```diff
@@ -148,6 +151,16 @@ private List<EpochEntry> removeWhileMatching(Iterator<Map.Entry<Integer, EpochEn
         return removedEpochs;
     }
 
+    public LeaderEpochFileCache writeTo(LeaderEpochCheckpoint leaderEpochCheckpoint) {
+        lock.writeLock().lock();
+        try {
+            leaderEpochCheckpoint.write(epochEntries());
```

Comment:
It seems like we are doing the following:
1. Flush the cache values to the checkpoint.
2. Create a new file cache which would read from the checkpoint created in step 1

which means, we are probably trying to have a copy of this cache backed by this new checkpoint here.

Alternatively, may I suggest the following:

```
public LeaderEpochFileCache cloneWithLeaderEpochCheckpoint(LeaderEpochCheckpoint leaderEpochCheckpoint) {
    this.flushToCheckpoint(leaderEpochCheckpoint);
    return new LeaderEpochFileCache(this.topicPartition, leaderEpochCheckpoint);
}
```
`flushToCheckpoint()` is similar to existing flush() method.

Reply:
Good suggestion!